### PR TITLE
support Debian non-AIO Puppet 4.x packages

### PR DESCRIPTION
**Pull request by mmoll**
@sathieu please test
4.5.2-1 packages are in the new queue for unstable at the moment, there were no changes that would affect this PR compared to 4.5.0-4.

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -63,7 +63,7 @@
     {
       "operatingsystem": "Fedora",
       "operatingsystemrelease": [
-        "21"
+        "24"
```
Comment (ekohl) [inline metadata.json]: This change looks unrelated, maybe this needs to be merged to master already?

File: manifests/params.pp [hunk 2]
```diff
@@ -100,7 +103,10 @@
         $bindir            = '/opt/puppetlabs/bin'
       } else {
         $dir               = '/etc/puppet'
-        $codedir           = '/etc/puppet'
+        $codedir           =  $deb_naio_package ? {
+                                true    => '/etc/puppet/code',
+                                default => '/etc/puppet',
```
Comment (domcleal) [inline manifests/params.pp]: Prefer "false" over "default", then if $deb_naio_package isn't set correctly then it should raise a compilation error.

File: manifests/params.pp [hunk 3]
```diff
@@ -258,7 +264,10 @@
   if $aio_package {
     $client_package = ['puppet-agent']
   } elsif ($::osfamily == 'Debian') {
-    $client_package = ['puppet-common','puppet']
+    $client_package = $deb_naio_package ? {
+      true    => ['puppet-agent','puppet'],
```
Comment (domcleal) [inline manifests/params.pp]: inconsistent spacing around `,` compared to the line below

File: metadata.json (hunk 1 above)
Comment (mmoll) [inline metadata.json]: done.

File: manifests/params.pp (hunk 2 above)
Comment (mmoll) [inline manifests/params.pp]: done

File: manifests/params.pp (hunk 3 above)
Comment (mmoll) [inline manifests/params.pp]: done

**Comment by mmoll** [comment]
anybody? :)

**Comment by domcleal** [comment]
Works well, merged.  Thanks @mmoll.
